src/geocat/f2py/checks.py

```diff
@@ -0,0 +1,164 @@
+import xarray as xr
+import numpy as np
+import dask as da
+
+def check(user_variable,
+          var_name='variable',
+          data_type=None,
+          max_dimensions=None,
+          min_dimensions=None,
+          dimensions=None,
+          shape=None,
+          is_xarray=None,
+          is_none=None,
+          comparison=None,
+          unchunked_dims=None):
+    '''
+    Function of data checks for use in wrapper functions.
+
+    user_variable : :class:`xarray.DataArray` or :class:`numpy.ndarray`:
+        Variable of interest
+
+    data_type : :class:`type` or :type:`tuple of types`:
+        Desired data_type of user_variable
+        ex: data_type=np.ndarray
+
+    max_dimensions : :class:`tuple`:
+        Maximum number of dimensions needed from user_variable for function
+        ex: min_dimensions=(3) or min_dimensions=3
+
+    min_dimensions : :class:`tuple`:
+        Minimum number of dimensions needed from user_variable for function
+        ex: min_dimensions=(3) or min_dimensions=3
+
+    dimensions : :class:`tuple`:
+        Number of dimensions needed from user_variable for function
+        ex: min_dimensions=(3) or min_dimensions=3
+
+    shape : :class:`tuple`:
+        Checks user_variable's shape
+        ex: shape=(3, 3, 3) or shape=(3,3,3)
+
+    is_xarray : :class:`bool`:
+        Checking that user_variable is of class xarray.DataArray.
+        Valid inputs are True or False
+
+    is_none : :class:`NoneType`:
+        Checking that user_variable is of class NoneType.
+        Valid input is None
+
+    comparison : :class:`int`:
+        Checks that user_variable is the correct type
+        ex: check(user_variable, comparison=
+
+    unchunked_dims : :class:`tuple`:
+        Checks for unchunked dimensions in user_variable
+        ex: check(user_variable, unchunked_dims=[3]) *must include brackets*
+
+    '''
+    if data_type is not None:
+        if not isinstance(user_variable, data_type):
+            raise TypeError(var_name + " failed data_type check.")
+
+    if dimensions is not None:
+        if not (user_variable.ndim == dimensions):
+            raise Exception(var_name + " failed dimensions check")
+
+    if shape is not None:
+        if not (user_variable.shape == shape):
+            raise Exception(var_name + " failed shape check")
+
+    if is_none is not None:
+        if is_none is True:
```

Comment:
Since `is_none` is already a boolean, there is no need for additional `is True`. This is also applicable to other blocks, if they written similarly. The whole block here could then be:

```
   if is_none is not None:
        if is_none:
            if user_variable is not None:
                raise Exception(var_name + " is not None")
        else:
            if user_variable is None:
                raise Exception(var_name + " is None")
```